Mathlib/FieldTheory/IntermediateField.lean

```diff
@@ -299,6 +297,23 @@ theorem toIntermediateField_toSubalgebra (S : IntermediateField K L) :
   rfl
 #align to_intermediate_field_to_subalgebra toIntermediateField_toSubalgebra
 
+/-- Turn an algebraic subalgebra into an intermediate field, `Subalgebra.IsAlgebraic` version. -/
+def Subalgebra.IsAlgebraic.toIntermediateField {S : Subalgebra K L} (hS : S.IsAlgebraic) :
+    IntermediateField K L where
+  toSubalgebra := S
+  inv_mem' x hx := Algebra.adjoin_le_iff.mpr
+    (Set.singleton_subset_iff.mpr hx) (hS x hx).isIntegral.inv_mem_adjoin
+
+/-- Turn an algebraic subalgebra into an intermediate field, `Algebra.IsAlgebraic` version. -/
+abbrev Algebra.IsAlgebraic.toIntermediateField (S : Subalgebra K L) [Algebra.IsAlgebraic K S] :
+    IntermediateField K L := (S.isAlgebraic_iff.mpr ‹_›).toIntermediateField
+
+/-- The algebraic closure of a field `K` in an extension `L`. Definitionally, this is exactly
+  `integralClosure K L` as an intermediate field. We choose the name `algClosure` to
+  distinguish the intermediate field from the subalgebra, and from `AlgebraicClosure`. -/
+def IntermediateField.algClosure : IntermediateField K L :=
```

Comment:
What about `IntermediateField.integralClosure`? The name `algClosure` looks a little strange.

Reply:
It's a bit strange to talk about the integral closure when you know it's a field. There are [AlgebraicClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.html#AlgebraicClosure) and [IsAlgClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/IsAlgClosed/Basic.html#IsAlgClosure) in mathlib, and I like the shorter name.

Reply:
In Mathlib, we have `separableClosure` as the relative separable closure (an intermediate field) and `SeperableClosure` as the absolute separable closure. And `normalClosure` for relative normal closure. Do you think `algebraicClosure` is a good name?

Reply:
Okay, even though there are IsSepClosed/Closure and IsAlgClosed/Closure, I think we should follow s/SeparableClosure and AlgebraicClosure here to name this algebraicClosure as you said.